src/geometry/PolySetUtils.cc

```diff
@@ -90,15 +94,15 @@ void tessellate_faces(const PolySet& inps, PolySet& outps)
 
   // Estimate how many polygons we will need and preallocate.
   // This is usually an undercount, but still prevents a lot of reallocations.
-  outps.polygons.reserve(polygons.size() );
+  PolySetBuilder builder(verts.size(),polygons.size());
+  for(int i=0;i<verts.size();i++)
+    builder.vertexIndex({verts[i][0],verts[i][1],verts[i][2]});
+
 
   for (const auto& faces : polygons) {
     if (faces[0].size() == 3) {
       // trivial case - triangles cannot be concave or have holes
-      outps.append_poly(3);
-      outps.append_vertex(verts[faces[0][0]]);
-      outps.append_vertex(verts[faces[0][1]]);
-      outps.append_vertex(verts[faces[0][2]]);
+       builder.append_poly({faces[0][0],faces[0][1],faces[0][2]});
```

Comment:
This function is *very problematic*. This will not preserve manifoldness because duplicated vertices are collapsed in the hash above, and added back without knowing any topological information.